### PR TITLE
TECH-134: Create test configuration for "/authorize" endpoint

### PR DESCRIPTION
**Pull request by KarolinaKopacz**
Create test configuration for "/authorize" endpoint.

Create test configuration for /authorize endpoint from https://github.com/GovStackWorkingGroup/bb-identity/pull/6 . Identify the critical APIs for Authentication, write the test plan and Cucumber tests, as well as supporting code to run those Cucumber tests.

TICKET: https://govstack-global.atlassian.net/browse/TECH-134

### CODE REVIEW

**Review by jdub13 — APPROVED** [review]

File: examples/mock/mockoon-bbidentity.json [hunk 1]
```diff
@@ -273,6 +273,108 @@
...
+          "headers": [],
+          "bodyType": "INLINE",
+          "filePath": "",
+          "databucketID": "",
+          "sendFileAsBody": false,
+          "rules": [
+            {
+              "target": "query",
+              "modifier": "scope",
+              "value": "^openid profile$|^openid$|^profile$|^email$|^address$|^phone$|^office_access$",
```
Comment (mgrochalskisoldevelo) [inline examples/mock/mockoon-bbidentity.json:298]: Looking at API should be offline_access - not office_access
image

File: examples/mock/mockoon-bbidentity.json [hunk 2]
```diff
@@ -273,6 +273,108 @@
...
```
Comment (mgrochalskisoldevelo) [inline examples/mock/mockoon-bbidentity.json:344]: the same as above - office_access -> offline_access

File: test/features/oidc_authorize.feature [hunk 3]
```diff
@@ -0,0 +1,36 @@
+Feature: The authorize endpoint of Open ID Connect (OIDC).
+
+  This is the authorize endpoint of Open ID Connect (OIDC), the relying party applications will do a browser redirect
+  to this endpoint with all required details passed as query parameters.
+
+  Request endpoint: GET /authorize
+
+  Scenario: Successfully loads JS application, and validates the provided query parameters using oauth-details endpoint
+    Given The user wants to loads JS application, and validates the provided query parameters using oauth-details endpoint
```
Comment (mgrochalskisoldevelo) [inline test/features/oidc_authorize.feature:9]: I think it should be ```The user wants to load JS application, and validates the provided query parameters using the oauth-details endpoint```

File: test/features/oidc_authorize.feature [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    When The user triggers an action with every required parameters
```
Comment (mgrochalskisoldevelo) [inline test/features/oidc_authorize.feature:10]: I think it should be ```The user triggers an action with every required parameter```

File: test/features/oidc_authorize.feature [hunk 5]
```diff
@@ -0,0 +1,36 @@
+Feature: The authorize endpoint of Open ID Connect (OIDC).
+
+  This is the authorize endpoint of Open ID Connect (OIDC), the relying party applications will do a browser redirect
+  to this endpoint with all required details passed as query parameters.
+
+  Request endpoint: GET /authorize
+
+  Scenario: Successfully loads JS application, and validates the provided query parameters using oauth-details endpoint
```
Comment (mgrochalskisoldevelo) [inline test/features/oidc_authorize.feature:8]: maybe ```using the oauth-details endpoint```

File: test/features/oidc_authorize.feature [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    Given The user wants to loads JS application, and validates the provided query parameters using oauth-details endpoint
+    When The user triggers an action with every required parameters
+    Then The user successfully loads JS application, and validates the provided query parameters using oauth-details endpoint
```
Comment (mgrochalskisoldevelo) [inline test/features/oidc_authorize.feature:11]: Same as the Scenario:

**Review by mgrochalskisoldevelo — CHANGES_REQUESTED** [review]
Please take a look at my comments - I mentioned just a few scenarios, the others have the same cases.

**Review by mgrochalskisoldevelo — APPROVED** [review]

**Review by dborowiecki — APPROVED** [review]
All necessary approvals were added, I'll merge the PR.

**Pull request merged**